File: skill_sdk/tests/fixtures/cpp-project/src/shop/FixedAmountDiscount.cpp
#include "FixedAmountDiscount.h"

FixedAmountDiscount::FixedAmountDiscount(double amount) : amount_(amount) {}

double FixedAmountDiscount::Apply(const Order& order) {
    return std::max(0.0, order.RawTotal() - amount_);
}

File: skill_sdk/tests/fixtures/js-project/src/shop/discount/FixedAmountDiscount.js
import { DiscountStrategy } from './DiscountStrategy.js';

/**
 * Subtracts a fixed amount from the total (floored at zero).
 *
 * @implements {DiscountStrategy}
 */
export class FixedAmountDiscount extends DiscountStrategy {
  /**
   * @param {number} amount
   */
  constructor(amount) {
    super();
    /** @type {number} */
    this._amount = amount;
  }

  /**
   * @param {import('../models/Order.js').Order} order
   * @returns {number}
   */
  apply(order) {
    return Math.max(0, order.rawTotal() - this._amount);
  }

  /** @returns {number} */
  getAmount() {
    return this._amount;
  }
}
